src/main/java/edu/brown/cs/student/main/Main.java

```diff
@@ -73,33 +76,175 @@ private void run() {
       runSparkServer((int) options.valueOf("port"));
     }
 
-    // TODO: Add your REPL here!
-    MathBot math = new MathBot();
+    // initialize a StarDistance object
+    StarDistance distCalculator = new StarDistance();
+
     try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
       String input;
+      // read the INPUT by line
       while ((input = br.readLine()) != null) {
         try {
           input = input.trim();
-          String[] arguments = input.split(" ");
-          // TODO: complete your REPL by adding commands for addition "add" and subtraction
-          //  "subtract"
-          if (arguments.length == 3 && this.isNumeric(arguments[1]) && this.isNumeric(arguments[2])) { // && arguments[1] & arguments[2] are both numeric)
-            double num1 = Double.parseDouble(arguments[1]);
-            double num2 = Double.parseDouble(arguments[2]);
-            if (arguments[0].equalsIgnoreCase("add")){
-              System.out.println(math.add(num1, num2));
+          String[] args = input.split(" ");
+
+          // hm2: {dist difference: starID[]}, a list of starIDs in case there are two stars at equal
+          // distance to the one given
+          HashMap<Double, ArrayList<String>> hm2;
+          // if the args have 2 values in total, the first one is "stars"
+          if (args.length == 2 && args[0].equals("stars")) {
+            // refresh stars array, counter, hashmap1 and hashmap3 so they are refreshed
+            stars = new ArrayList<>();
+            counter = 0;
+            hm1 = new HashMap<>();
```

Comment:
I would rename the hashmaps more descriptively to make things more readable to you and others